pycodestyle.py

```diff
@@ -2125,21 +2125,30 @@ def generate_tokens(self):
             self.report_error(1, 0, 'E902 %s' % self._io_error, readlines)
         tokengen = tokenize.generate_tokens(self.readline)
         try:
+            prev_physical = ''
             for token in tokengen:
                 if token[2][0] > self.total_lines:
                     return
                 self.noqa = token[4] and noqa(token[4])
-                self.maybe_check_physical(token)
+                self.maybe_check_physical(token, prev_physical)
                 yield token
+                prev_physical = token[4]
         except (SyntaxError, tokenize.TokenError):
             self.report_invalid_syntax()
 
-    def maybe_check_physical(self, token):
+    def maybe_check_physical(self, token, prev_physical):
```

Comment:
We may need to replicate some of this logic to Flake8

Reply:
indeed, I plan to work on that today!

Reply:
here's the flake8 patch with essentially the same logic!

https://gitlab.com/pycqa/flake8/-/merge_requests/451